File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/relatorio/DemonstrativoAtualizacaoJRAdapter.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio;

import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapter;
import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapterDataSource;
import br.jus.trt8.pjecalc.base.comum.relatorio.JREmptyDS;
import java.math.BigDecimal;

public abstract class DemonstrativoAtualizacaoJRAdapter
extends JRAdapter {
    public abstract JRAdapterDataSource<DemonstrativoAtualizacaoAdapter> getCreditoDebitoOutroDebito();

    public abstract JRAdapterDataSource<JREmptyDS> getEmptyDS();

    public abstract class DemonstrativoItemAdapter
    extends JRAdapter {
        public abstract String getDescricao();

        public abstract String getBase();

        public abstract String getTaxa();

        public abstract String getValor();

        public abstract String getIndice();

        public abstract String getDevido();

        public abstract String getPago();

        public abstract String getDiferenca();
    }

    public abstract class DemonstrativoAtualizacaoAdapter
    extends JRAdapter {
        public abstract String getDescritivoDeEventos();

        public abstract String getCabecalho();

        public abstract BigDecimal getTotalDevido();

        public abstract BigDecimal getTotalPago();

        public abstract BigDecimal getTotalDiferenca();

        public abstract JRAdapterDataSource<DemonstrativoItemAdapter> getItens();
    }

    public abstract class Item {
    }
}

File: src/lib/pjecalc/core/dominio/calculo/relatorio/demonstrativo-atualizacao-jr-adapter.ts
/**
 * PJe-Calc v2.15.1 — DemonstrativoAtualizacaoJRAdapter (abstract)
 * Porte TS-adaptado de:
 *   br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.DemonstrativoAtualizacaoJRAdapter
 *
 * Ref Java: pjecalc-fonte/.../relatorio/DemonstrativoAtualizacaoJRAdapter.java
 *
 * Demonstrativo pós-liquidação mostrando, por evento de atualização, os
 * valores atualizados dos créditos/débitos do reclamante.
 */
import type { Periodo } from '../../../base/comum/periodo';
import type Decimal from 'decimal.js';
import { JRAdapter, JRAdapterDataSource } from './jr-adapter';

export abstract class DemonstrativoAtualizacaoJRAdapter extends JRAdapter {
  abstract getEventos(): JRAdapterDataSource<DemonstrativoAtualizacaoEventoAdapter>;
  abstract getPeriodo(): Periodo;
  abstract getTotalAtualizado(): Decimal;
}

export abstract class DemonstrativoAtualizacaoEventoAdapter extends JRAdapter {
  abstract getDataEvento(): Date;
  abstract getDescricaoEvento(): string;
  abstract getValorOriginal(): Decimal;
  abstract getIndiceCorrecao(): Decimal;
  abstract getValorCorrigido(): Decimal;
  abstract getJuros(): Decimal;
  abstract getTotal(): Decimal;
  abstract getPago(): Decimal;
  abstract getDiferenca(): Decimal;
}
